### PR TITLE
 SDKQE-2501: Add search index management tests

### PR DESCRIPTION
**Pull request by jrawsthorne**

### CODE REVIEW

File: test/test_integration_management.cxx [hunk 1]
```diff
@@ -1004,3 +1005,252 @@ TEST_CASE("integration: query index management", "[integration]")
...
+    {
+        couchbase::operations::management::search_index index;
+        index.type = "fulltext-index";
+        index.source_type = "couchbase";
+        index.source_name = integration.ctx.bucket;
+        couchbase::operations::management::search_index_upsert_request req{};
+        req.index = index;
+        auto resp = test::utils::execute(integration.cluster, req);
+        REQUIRE(resp.ctx.ec == couchbase::error::common_errc::invalid_argument);
+    }
```
Comment (jrawsthorne) [inline test/test_integration_management.cxx:1134]: gocb has this test. Do we want to return early with invalid_argument if the name is empty?

Comment (avsej) [reply test/test_integration_management.cxx:1134]: yes, I think it makes sense. Just like with analytics links we can validate it and return error. I can add the fix for it

**Comment by brett19** [comment]
test

**Comment by brett19** [comment]
run tests

**Comment by brett19** [comment]
test this please

**Comment by brett19** [comment]
test this please

**Comment by brett19** [comment]
test this please

**Comment by brett19** [comment]
test this please

**Comment by brett19** [comment]
test this please

**Review by avsej — APPROVED** [review]

**Pull request merged**